### PR TITLE
User update details

### LINKED ISSUE #65: Add columns to user model
### Feature Categories:

- [x] Express
- [ ] React
- [ ] Styling/CSS
- [ ] Documentation
- [ ] Other: 

**Describe the feature/Bug your gonna work on**
Add information to users table. 
- [x] image
- [x] bio
- [x] street 
- [x] city
- [x] state 
- [x] zip

**Additional context**
Add any other context or screenshots about the feature.

### PR DESCRIPTION
**Pull request by Strompy**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added function to update the users' details via graphQL. Make sure to update your database with the new migrations and seeds.
userUpdate Mutation is now available on GraphQL. Requires the user's ID to be passed along with the updated data.

database.json has been added to the gitignore, so that local user changes aren't being passed with every PR

closes #66  

`Your message here`
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [X] No
- [ ] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
Tests to come
# Message for reviewer:

# Checklist:
- [X] My code has no unused/commented out code
- [X] My code has no console logs
- [X] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/api/src/modules/subscription/resolvers.js [hunk 1]
```diff
@@ -58,4 +58,4 @@ export async function remove(parentValue, { id }, { auth }) {
   } else {
     throw new Error('Access denied.')
   }
-}
\ No newline at end of file
+}
```
Comment (Abdeboskey) [inline code/api/src/modules/subscription/resolvers.js:61]: Y'all get to write those really cool robo-errors.

File: code/api/src/modules/user/query.js [hunk 2]
```diff
@@ -34,36 +34,6 @@ export const userLogin = {
...
-      name: 'bio',
-      type: GraphQLString
-    },
-
-    image: {
-      name: 'image',
-      type: GraphQLString
-    },
-
     role: {
```
Comment (Abdeboskey) [inline code/api/src/modules/user/query.js:37]: Beautiful. Thank you!

**Review by Abdeboskey — APPROVED** [review]
This looks awesome, y'all are on a roll. Thank you!! I appreciate y'all being so considerate of us when making these changes!

File: code/api/src/modules/subscription/resolvers.js (hunk 1 above)
Comment (Abdeboskey) [reply code/api/src/modules/subscription/resolvers.js:61]: `ACCESS DENIED`

File: code/api/src/modules/user/mutations.js [hunk 3]
```diff
@@ -49,12 +66,17 @@ export const userSignup = {
...
+      type: GraphQLString
+    },
+
+    image: {
+      name: 'image',
       type: GraphQLString
     }
   },
-  resolve: create
+  resolve: update
```
Comment (rwilliams659) [inline code/api/src/modules/user/mutations.js:79]: Stoked to have this implemented!

File: code/api/src/modules/user/resolvers.js [hunk 4]
```diff
@@ -64,6 +65,28 @@ export async function login(parentValue, { email, password }) {
   }
 }
 
+// Update
+export async function update(parentValue, { id, name, email, street, city, state, zip, bio, image }, { auth }) {
+  console.log(auth.user.id)
```
Comment (rwilliams659) [inline code/api/src/modules/user/resolvers.js]: NBD, but just pointing out this console.log is still here

**Review by rwilliams659 — APPROVED** [review]
Great job y'all!

File: code/api/.gitignore [hunk 5]
```diff
@@ -2,3 +2,4 @@ node_modules/
 build/
 public/images/uploads/*
 !public/images/uploads/.gitkeep
+src/config/database.json
```
Comment (sienna-kopf) [inline code/api/.gitignore:5]: ⭐️  ing this to make sure we include it in the READme.

File: code/api/src/modules/user/resolvers.js [hunk 6]
```diff
@@ -64,6 +65,28 @@ export async function login(parentValue, { email, password }) {
   }
 }
 
+// Update
+export async function update(parentValue, { id, name, email, street, city, state, zip, bio, image }, { auth }) {
```
Comment (sienna-kopf) [inline code/api/src/modules/user/resolvers.js:69]: Figuring out how a mutation takes and uses arguments was really cool!

File: code/api/src/modules/user/resolvers.js (hunk 4 above)
Comment (sienna-kopf) [reply code/api/src/modules/user/resolvers.js]: Good point @ rachel. I'll pull this down, remove it and push it back up to this PR

File: code/api/src/modules/user/resolvers.js [hunk 7]
```diff
@@ -64,6 +65,28 @@ export async function login(parentValue, { email, password }) {
...
+        name,
+        email,
+        bio,
+        image,
+        street,
+        city,
+        state,
+        zip
+      },
+      {where: { id } }
```
Comment (sienna-kopf) [inline code/api/src/modules/user/resolvers.js:82]: So fascinating to me that id has to be in an object wrapper for the sequelize where clause. Something to keep in mind when we jump into order history.

File: code/api/src/modules/user/mutations.js [hunk 8]
```diff
@@ -49,12 +66,17 @@ export const userSignup = {
       type: GraphQLString
     },
 
-    password: {
-      name: 'password',
+    bio: {
```
Comment (JoshSevy) [inline code/api/src/modules/user/mutations.js:69]: Lovely, just lovely. Excited to add this to the front end.

**Review by JoshSevy — APPROVED** [review]
Great work Team. You all are on a roll. So excited to have you all on the same team.

**Comment by Strompy** [comment]
Pulled out that errant console.log. Yay Javascript!

**Pull request merged**